rflx/model.py

```diff
@@ -1239,6 +1247,39 @@ def __prove_field_positions(self) -> None:
                     )
                     return
 
+                if f in self.__types and isinstance(self.__types[f], Opaque):
+                    start_aligned = Not(
+                        Equal(Mod(self.__target_first(last), Number(8)), Number(1), last.location)
+                    )
+                    proof = start_aligned.check([*facts, *self.__type_constraints(start_aligned)])
+                    if proof.result != ProofResult.unsat:
+                        path_message = " -> ".join([p.target.name for p in path])
+                        self.error.append(
+                            f'opaque field "{f.name}" not aligned to 8 bit boundary'
+                            f" ({path_message})",
+                            Subsystem.MODEL,
+                            Severity.ERROR,
+                            self.location,
+                        )
+                        return
+
+                    length_multiple_8 = Not(
+                        Equal(Mod(self.__target_length(last), Number(8)), Number(0), last.location)
+                    )
+                    proof = length_multiple_8.check(
+                        [*facts, *self.__type_constraints(length_multiple_8)]
```

Comment:
Do we want to hard code 8 or some type of byte length instead?

Reply:
I think using `Opaque.element_size` (which itself is just `Number(8)`) would be a good choice. Even though I don't assume we'll ever change that size, this solution is cleaner. Done in 33228ce.